src/config.ts

```diff
@@ -17,20 +17,17 @@ export const findOrCreateConfig = async (config?: string) => {
 	// Try to locate a config file in the current working directory, or with the `config` option if it was provided.
 	const configSource = config ?? (await searchForConfig(process.cwd()))?.__importRef?.filename;
 
-	// LocalConfigPath is a path to a config file in the user's configuration directory (platform dependent).
-	const defaultConfigPath = findDefaultConfigPath('cspell.json');
-
 	// If no config file was found, use/create a config file in the user's configuration directory (platform dependent).
-	const path = configSource ?? defaultConfigPath;
+	const path = configSource ?? findDefaultConfigPath('cspell.json');
 
 	// Only JSON files are supported to prevent more dependencies for yml parsing. If the config file is not a JSON
 	// file, it can still be used, but it won't be updated with new ignored words.
 	if (extname(path) === '.json') {
 		configPath = path;
 	}
 
-	// If the path is the local config path and it doesn't exist, create it.
-	if (path === defaultConfigPath && !existsSync(defaultConfigPath)) {
+	// If configSource is undefined, check if default config path exists. If it doesn't, create it.
+	if (!configSource && !existsSync(path)) {
 		await writeToSettings({});
```

Comment:
I think that technically, the following would be the most optimized:

```ts
	if (configSource) {
		if (extname(path) === '.json') {
			configPath = path;
		}
	} else if (!existsSync(path)) {
		await writeToSettings({});
	}
```

Which wouldn't call `extname` if it was the default path. But I'm even annoying myself now, so it's fine 🤣